File: 1828.py
def jogada():
    s, r = map(str,input().split())
    return s,r


def contagem(cont):
    cont += 1
    return cont


def winner(a,b,cont,play):
    w = 1
    if a == 'tesoura' and b == 'papel':
        w = a
    if a == 'papel' and b == 'pedra':
        w = a
    if a == 'pedra' and b == 'lagarto':
        w = a
    if a == 'lagarto' and b =='Spock':
        w = a
    if a == 'Spock' and b == 'tesoura':
        w = a
    if a == 'tesoura' and b == 'lagarto':
        w = a
    if a == 'lagarto' and b == 'papel':
        w = a
    if a == 'papel' and b == 'Spock':
        w = a
    if a == 'Spock' and b == 'pedra':
        w = a
    if a == 'pedra' and b == 'tesoura':
        w = a
    if w == a:
        if play == 'she':
            print('Caso #{}: Bazinga!'.format(cont))
        if play == 'raj':
            print('Caso #{}: Raj trapaceou!'.format(cont))
    return w


def main():
    n = int(input())
    cont = 1
    for i in range(n):
        s, r = jogada()
        w = winner(s,r,cont,'she')
        if w != 1:
            cont = contagem(cont)
            continue
        
        w = winner(r,s,cont,'raj')
        if w != 1:
            cont = contagem(cont)
            continue
        
        if w == 1:
            print('Caso #{}: De novo!'.format(cont))
        cont = contagem(cont)
        
main()

File: CSharp/1828.cs
using System;
using System.Collections.Generic;

class URI
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int cont = 1;
        
        for (int i = 0; i < n; i++)
        {
            string[] input = Console.ReadLine().Split();
            string s = input[0];
            string r = input[1];

            string w = Win(s, r, cont, "she");
            if (w != "1")
            {
                cont = Contagem(cont);
                continue;
            }

            w = Win(r, s, cont, "raj");
            if (w != "1")
            {
                cont = Contagem(cont);
                continue;
            }

            if (w == "1")
            {
                Console.WriteLine($"Caso #{cont}: De novo!");
            }
            cont = Contagem(cont);
        }
    }

    static int Contagem(int cont)
    {
        cont += 1;
        return cont;
    }

    static string Win(string a, string b, int cont, string quem)
    {
        string w = "1";
        if (a == "tesoura" && b == "papel") w = a;
        if (a == "papel" && b == "pedra") w = a;
        if (a == "pedra" && b == "lagarto") w = a;
        if (a == "lagarto" && b == "Spock") w = a;
        if (a == "Spock" && b == "tesoura") w = a;
        if (a == "tesoura" && b == "lagarto") w = a;
        if (a == "lagarto" && b == "papel") w = a;
        if (a == "papel" && b == "Spock") w = a;
        if (a == "Spock" && b == "pedra") w = a;
        if (a == "pedra" && b == "tesoura") w = a;

        if (w == a)
        {
            if (quem == "she")
            {
                Console.WriteLine($"Caso #{cont}: Bazinga!");
            }
            else if (quem == "raj")
            {
                Console.WriteLine($"Caso #{cont}: Raj trapaceou!");
            }
        }
        return w;
    }
}
